server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/DatasetService.java

```diff
@@ -256,10 +257,12 @@ private DatasetInfoVo toDatasetInfoVo(DatasetEntity ds, DatasetVersionEntity ver
         }
     }
 
-    public void shareDatasetVersion(String projectUrl, String datasetUrl, String versionUrl,
-            Boolean shared) {
-        Long versionId = bundleManager.getBundleVersionId(BundleVersionUrl
-                .create(projectUrl, datasetUrl, versionUrl));
+    public void shareDatasetVersion(String projectUrl, String datasetUrl, String versionUrl, Boolean shared) {
+        var project = projectService.getProjectVo(projectUrl);
+        if (!project.getPrivacy().equals(Project.Privacy.PUBLIC.name())) {
```

Comment:
if (**shared &&** !project.getPrivacy().equals(Project.Privacy.PUBLIC.name())) ?

Reply:
We will make sure that the private project never has a shared resource